oci-unit-tests/helper/test_helper.sh

```diff
@@ -97,3 +100,61 @@ check_manifest_exists()
     debug "not found"
     return 1
 }
+
+# Check that the images listed
+# $1: List of all supported architectures to check.
+#     If empty, defaults to $SUPPORTED_ARCHITECTURES.
+check_supported_architectures()
+{
+    local supported_architectures="$1"
+    local registry
+    local utilsdir
+    local listed_architectures
+    local ret
+    ret=0
+
+    # We have to massage the registry name a bit.
+    case "${DOCKER_REGISTRY}" in
+	*"docker"*)
+	    debug "This function does not currently support the docker.io registry."
+	    return 0
+	    ;;
+	*"aws"*)
+	    registry="aws"
+	    ;;
+    esac
+
+    utilsdir="$(mktemp -d)/utils"
+    trap 'rm -rf ${utilsdir}' 0 INT QUIT ABRT PIPE TERM
+
+    debug -n "Checking if all supported architectures are present in the Docker manifest file"
+
+    if [ -z "${supported_architectures}" ]; then
+	# shellcheck disable=SC2153
+	supported_architectures="${SUPPORTED_ARCHITECTURES}"
+    fi
+
+    if ! git clone -q --depth 1 https://git.launchpad.net/~canonical-server/ubuntu-docker-images/+git/utils "${utilsdir}"; then
+	debug "failed to clone 'utils' repository"
+	rm -rf "${utilsdir}"
+	return 1
+    fi
+
+    listed_architectures=$("${utilsdir}"/list-manifest-for-image-and-tag.sh \
+					--registry "${registry}" \
+					--namespace "${DOCKER_NAMESPACE}" \
+					--image "${DOCKER_PACKAGE}" \
+					--tag "${DOCKER_TAG}" \
+			       | jq -r '.manifests[].platform.architecture')
+
```

Comment:
General comment: there is a mix of spaces and tabs in this change (and in  part in the already committed scripts). It's not stated anywhere, but ~all the other scripts in this repo are using a 4-space indent, no tabs.

/bikeshed

Reply:
Ah, fair enough.  I think I'm the only one using Emacs here :-).  I'll fix the indentation.